pekko-stream/src/main/scala/eu/neverblink/jelly/pekko/stream/impl/JellyIoOps.scala

```diff
@@ -48,6 +52,50 @@ object JellyIoOps:
     final def fromBytes: Flow[Array[Byte], RdfStreamFrame, NotUsed] =
       Flow[Array[Byte]].map(RdfStreamFrame.parseFrom)
 
+    /**
+     * Convert a stream of DELIMITED bytes into a stream of Jelly frames. The ByteStrings may be chunked
+     * in an arbitrary way, the stream will be framed based on the Protobuf varint-encoded length prefix.
+     *
+     * For example, one input ByteString may contain multiple Jelly frames, a part of a Jelly frame, or
+     * any combination of these.
+     *
+     * Using this method, you can read Jelly files from a file or socket in a fully reactive manner,
+     * without any blocking operations.
+     *
+     * @return Pekko Flow
+     */
+    final def fromByteStreamDelimited: Flow[ByteString, RdfStreamFrame, NotUsed] =
+      Flow[ByteString]
+        .via(protobufFraming(Int.MaxValue))
+        .map(byteString => {
+          val byteStrings = byteString.asByteBuffers
+          val codedInputStream = if byteStrings.size > 1 then
+            CodedInputStream.newInstance(byteStrings.map(_.asReadOnlyBuffer()).asJava)
```

Comment:
Aren't I converting the ByteStrings to ByteBuffers twice here? Double check it.

Reply:
yeah, isnt it like.. the same in terms of what it does?
CodedInputStream.newInstance has different behaviour for collections of byte arrays and just byte array?

Reply:
It actually does. When called with just one ByteBuffer, it applies extra optimizations. I'll add a comment to clarify this, thanks for pointing it out.